### ISSUE TITLE
Document changes between Autokey and AutoHotKey (Windows)

### ISSUE THREAD

**Issue by jose1711** [autokey/autokey#643]
## Classification:
UI/Usability

## Reproducibility:
n/a

## AutoKey version:
irrelevant

## Used GUI:
irrelevant

## Installed via:
irrelevant

## Linux distribution:
any

## Summary:
For AutoHotKey (AHK) users coming from Windows looking for a similar alternative in Linux a comparison between AHK and AutoKey would be welcome. At this moment all I was able to find about AutoKey was that it provides a subset of features provided by AHK. While I think it's quite the opposite (AutoKey backed by Python has much greater potential) it would be nice if there is something like a side-by-side comparison from users with experience from both programs.

## Steps to reproduce: 
Try to find comparison between AHK and AK when it comes to features.

## Expected result:
Success.

## Actual result:
Nothing.

**Comment by josephj11** [comment]
The easiest way to approach this would be (for you to) create a page in our Wiki (anyone with a GitHub account can do this) and start an outline or some sort of list. Then, anyone who has something to contribute can jump in and add to it.

GitHub Markdown takes a little bit of effort to get used to, but it's pretty simple to use for most things.

AutoKey was inspired by Phrase Express, not by AutoHotKey, so there are no direct one-for-one correspondences.

Yes, we believe AutoKey is more powerful than AutoHotKey. While AutoHotKey has some shortcuts to getting things done that we do not, the availability of the full power of Python for scripting provides far greater power than the domain specific language AutoHotKey provides while having the advantage of being one of the most popular languages that many people are already familiar with.

**Comment by Elliria** [comment]
I took a quick look at AutoHotKey and, like AutoKey, AutoHotKey's strength seems to be in scripted actions that can be automated. One difference between the two that immediately caught my eye is that AutoHotKey uses its own scripting language, whereas AutoKey uses Python.

It would probably be impossible to do a straight-across comparison of the two, because AutoKey can do anything you can script with Python, which is pretty much anything you can imagine. You can even run your existing Python scripts from within AutoKey. The phrases and scripts that AutoKey uses can be manually triggered or can be triggered automatically and you can customize those triggers. It would probably take a lifetime to list off everything you can do with it. Once you get started, you'll find yourself on an endless journey of exploration to find out what you can get it to do next.

I don't know what the AutoHotKey community is like, but we welcome conversations about AutoKey and contributions of code and/or example scripts and/or documentation. Feel free to participate as much as you like. Your efforts will be appreciated. Also, if you join us in any of our other platforms (like Google or Gitter), hopefully you'll find us to be a friendly and helpful bunch. If you don't get an immediate response, please don't give up. We all come poking around in the various AutoKey spots every now and again and someone will surely answer eventually.

Last, but not least, if you have specific types of things that you currently do with AutoHotKey and are wondering if you'll be able to do those in AutoKey, those would make good, specific questions and we may already have scripts that can do them. I agree with Joe that the [AutoKey wiki](https://github.com/autokey/autokey/wiki) is a great place to start, and I think you'll find the **Features** section especially interesting. You'll find example scripts and a whole lot more there.

By the way, when creating AutoKey issues, whenever you have a question (as opposed to an issue), it's better to post it on our [Google list](https://groups.google.com/forum/#!forum/autokey-users) or in our [Gitter chat](https://gitter.im/autokey/autokey). The list is more user-oriented, whereas more technical questions will get a better answer on Gitter. Both of these venues have a number of active users, so you're likely to get more, better, and faster answers on them than in here on GitHub, where only our very busy developers and a few users see the issue reports. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user-community could handle for them.

**Comment by josephj11** [comment]
I do think this is a valid issue because it is requesting additional/enhanced documentation rather than "How do I do this?"

**Comment by Elliria** [comment]
Oh, sorry about that. I didn't mean to come across as hard-nosed or anything.

**Comment by josephj11** [comment]
i don't think you did, but you led with ask elsewhere, so it might be construed as "you should have done this instead".
I usually put that at the end of a post with the BTW in front of it to avoid/soften this impression.

**Comment by Elliria** [comment]
Good point. I edited it and will try to remember that in the future. Thanks for the help.

**Comment by kreezxil** [comment]
I agree with @Elliria the two macro/scripting programs while the most popular on their platforms are nothing alike, and yet have the same strengths. Comparing the two would be the same as comparing AHK script to Python, literally. 

Moreover, in as much as I've contributed here now as 4 different handles over the course of the project's lifetime, I've contributed on AHK too. They, last I checked, are nearly as warm as we are.

**Comment by josephj11** [comment]
Rather than a feature comparison or formal document, maybe what we need is a short essay from someone who knows both on the differences in philosophy and general approaches to using both programs. Their phrase capabilities seem pretty similar, but the scripts are worlds apart.

If this were written well enough, it could possibly be published elsewhere so that a larger audience would see it.

**Comment by kreezxil** [comment]
Sounds like a good idea. I've recently had the need to set up a proper
Windows 10 partition, I'm trying desperately how to figure out a way to get
my scripts to server me on that side. I'm thinking of maybe just coming up
with a bridge between AHK and python to do it. Furthermore, I know that's
simplistic, there might be another way.

**Comment by kreezxil** [comment]
Oh wow, look at what I just found!

https://pypi.org/project/macpy/

Says it supports everything we do now, but under both Linux and Windows. Considering it's another opensource program, maybe we can use it to bring AUTOKEY to windows.

I don't know much about but I'm gonna try. :)

**Comment by josephj11** [comment]
@kreezxil  I looked at that project briefly. Appears to be an API similar to what the AutoKey API offers (but without all the front end and application stuff AutoKey provides, so you have to roll your own actual application to use it).

This brings a couple of related issues to mind.

AFAIK, the AutoKey Record functionality has been broken for a very long time. If that's still the case, then our devs might find some solutions by looking at the way this project works or possibly use it to replace our Record code.

As for actually solving

Also, they claim to have limited support for Wayland. Maybe we can also learn something about this by studying their code.

As for actually solving your cross platform issue, that takes me back to my soapbox (mentioned on our Discord channel) about putting a small processor between your keyboard/mouse and computer. Such a processor would do whatever you like regardless of what's on the computer receiving its output although things like window manipulation would be a problem.
